tests/test_variant_algebra.py

```diff
@@ -124,6 +124,50 @@ def test_ordering_with_tail():
     assert res["cuda_compiler_version"] == ["None", "12.9"]
 
 
+@pytest.mark.parametrize("gcc_for_12dot6", ["13", "14", "15"])
+def test_ordering_with_primary_key(gcc_for_12dot6):
+    # ensure that GCC pins matching the respective CUDA versions get merged
+    # correctly; the GCC pin for 12.6 should never get picked since that CUDA
+    # version gets dropped in the merge of the primary keys (based on ordering).
```

Comment:
Can you remind me why a variant add operation drops a version instead of simply adding a version on top?

Reply:
The thing is, we're not in a variant-add operation; those get handled further up, i.e.
https://github.com/conda-forge/conda-smithy/blob/482e2de0d2f0617a9b3e8ae58ed5270b14d3832c/conda_smithy/variant_algebra.py#L308-L310

I don't know the origin/reasons behind the naming of the functions, but as the test shows (not least by lack of absence of `operation: key_add`), we're doing a normal "update-one-value-with-another" migrator.

The particularities are that we have two values per key, and they get merged with an `ordering:`. The ordering for the `primary_key:` must then be respected by the other keys in the zip (this is the core of the PR), otherwise the merged values may be jumbled completely (e.g. not form valid variants per set of values).